tests/production/AddUserSettingsToCouchTests.js

```diff
@@ -0,0 +1,122 @@
+/**
+Tests for adding test gpii keys and 'user' prefs safe.
+
+Copyright 2019 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii");
+
+fluid.require("%gpii-universal");
+
+gpii.loadTestingSupport();
+
+fluid.registerNamespace("gpii.tests.productionConfigTesting");
+
+require("./Common.js");
+
+// Tests that add 'user' preferences to the data base
+fluid.defaults("gpii.tests.productionConfigTesting.addUserSettings", {
+    gradeNames: ["fluid.test.sequenceElement"],
+    sequence: [
+        { funcName: "fluid.log", args: ["Add 'user' test prefs safes tests:"]},
+        {
+            func: "{addSettingsUserKey}.send",
+            args: [
+                gpii.tests.productionConfigTesting.settingsUserKey,
+                { port: "5984" }
```

Comment:
Is it necessary to specify "port" as it has been defined in the request at line 82? The same comment applies to other places too.

Reply:
Yes, it's necessary.  As far back as November, setting the port in the "testdef" section has no effect if the configuration file has a "mainServerPort" property. 

Initially, the problem was evident when specifying 9082 for the flowmanager port in the testDef only to have 8081 used consistently instead.  8081 was used because the "mainServerPort" was given as 8081.  I modified the production config file to use 9082, and all was well ... for a time.

The same problem cropped up when adding/deleting test records to/from the database using port 5984.  Specifying the port as 5984 in the testDef had no effect -- 9082 was always used instead.  Specifying the port in the `directModel` parameter of the `send()` function fixes it.  I left the port specified in the testDef as documentation, but I could take it out of there.